pkg/meta/meta.go

```diff
@@ -1560,7 +1576,50 @@ func (m *Meta) GetHistoryDDLJobsIterator(startJobID int64) (LastJobIterator, err
 
 // HLastJobIterator is the iterator for gets the latest history.
 type HLastJobIterator struct {
-	iter *structure.ReverseHashIterator
+	iter       *structure.ReverseHashIterator
+	schemaIDs  set.Int64Set
+	tableNames set.StringSet
+}
+
+// extractSchemaIDAndTableName extract schema_id and table_id from encoded Job structure
+// Note, here we strongly rely on the order of fields in marshalled string, just like checkSubstringsInOrder
+func extractSchemaIDAndTableName(s string) (int64, string, error) {
+	pos := strings.Index(s, `"schema_id":`)
+	if pos == -1 {
+		return 0, "", fmt.Errorf("schema_id not found in model.Job json")
+	}
+
+	start := pos + len(`"schema_id":`)
+	substr := s[start:]
+	end := strings.Index(substr, ",")
+	if end == -1 {
+		end = len(substr)
+	}
+
+	schemaIDStr := strings.TrimSpace(substr[:end])
+	schemaID, err := strconv.ParseInt(schemaIDStr, 10, 64)
+	if err != nil {
+		return 0, "", fmt.Errorf("failed to parse table_id: %v", err)
+	}
+
+	pos = strings.Index(substr, `"table_name":`)
+	if pos == -1 {
+		return 0, "", fmt.Errorf("table_name not found in model.Job json")
+	}
+
+	start = pos + len(`"table_name":`)
+	substr = substr[start:]
+	end = strings.Index(substr, ",")
```

Comment:
the table name may contain `,`. Please add unit test to cover these corner cases:

- table name is `` `t1,2` ``
- `schema_id` and `table_name` appears in column names, table comments and many places

Reply:
- I wonder when will the `table_name` contains comma, just curious.
- Seems like `table_name` and `schema_name` only appear once. 
https://github.com/pingcap/tidb/blob/eaa75a81d56471b602efa3f4ee19bc5345f2cdff/pkg/parser/model/ddl.go#L490-L501
BTW, I'll add tests to cover these cases.

Reply:
> I wonder when will the table_name contains comma, just curious.

It's decided bu the user. Maybe user will create such table.

> Seems like table_name and schema_name only appear once.

It mean appears in column names, table comments and many places. For example, `CREATE TABLE ... COMMENT 'table_name'`